Issue #66 - Form makeover

EChesters opened the pull request:
@womenhackfornonprofits/whfnp

[Issue-77](https://github.com/womenhackfornonprofits/whfnp/issues/77)
-  button sizes reconstructured
-  Form classes to use ones from Bootstrap
-  Added new colour for help-text
-  Added accessibility required aria attributes
-  Increased font-weight so it's readable and changed colours to black rather than red

EChesters committed:
Issue #66 - Form makeover
+  button sizes reconstructured
+  Form classes to use ones from Bootstrap
+  Added new colour for help-text
+  Added accessibility required aria attributes
+  Increased font-weight so it's readable and changed colours to black rather than red

EChesters commented:
_This is actually Issue #113 and not Issue 66_ I saw the wrong issue when I was typing the commit message and I can't change it.

EChesters committed:
Issue #77 - Fixed forms on volunteer and non-profits pages

EChesters commented:
Volunteer contact form:
<img width="512" alt="screen shot 2016-06-19 at 21 18 42" src="https://cloud.githubusercontent.com/assets/2605703/16179648/6dbe5598-3663-11e6-8fef-ff39726af5ca.png">

NP contact form:
<img width="525" alt="screen shot 2016-06-19 at 21 17 59" src="https://cloud.githubusercontent.com/assets/2605703/16179649/6dc26430-3663-11e6-8354-c892bacd015e.png">

Contact form:
<img width="903" alt="screen shot 2016-06-19 at 20 03 28" src="https://cloud.githubusercontent.com/assets/2605703/16179650/6dc983be-3663-11e6-86fd-8401d51b76ba.png">

EChesters committed:
Added a space for first name

On all the other forms we ask for a contact name or just name.

EChesters commented:
In terms of the names fields I have some questions:
1. Is there any reason why we separate them?
2. Will there be any back end issue not making surname a required field - not everyone in this world has a surname. 

_Can we remove the surname field and combine it with first name, or just remove the required attribute?_

Nandhini31 commented:
@EChesters :+1: Yes fine with all the new forms. happy to have just one name field.

tanyapowell commented on src/contact.html:
```diff
@@ -7,20 +7,32 @@
-		<textarea name="message" rows="5" placeholder="E.g. Hello, just wanted to tell you I think you are awesome" required></textarea>
-		<input type="submit" value="Send">
+		<div class="form-group">
+	    	<label for="name">Name</label>
```
This is not specifically for this line, but generally for throughout this PR (wanted to use this line as an example though) 
- Wrap radio and checkbox inputs and their text in labels. No need for `for` attributes. The wrapping automatically associates the two.

```
<!-- bad -->
<label for="email">
<input id="email" name="email">

<!-- good -->
<label>Name: <input name="email"></label>
```

EChesters replied:
Tried this, and it broke Bootstrap's styling. This was decided with @tanyapowell to try at a later stage.

tanyapowell commented on src/contact.html:
```diff
@@ -7,20 +7,32 @@
+	    	<label for="name">Name</label>
+			<input type="text" name="name" placeholder="Ada Lovelace" required aria-required="true" class="form-control"/>
+		</div>
+		<div class="form-group">
```
Really we should avoid the extra overhead of having the `label` and `input` tag inside a parent `div`. Is this a bootstap thing why each `label`/`input` tag inside a parent `div`?

EChesters replied:
This is for Bootstrap styling, although I agree with you. I'm not sure why Bootstrap do this. See on here for their example: http://getbootstrap.com/css/#forms

tanyapowell commented on src/sass/partials/_button.scss:
```diff
@@ -1,26 +1,34 @@
 	}
 	
-}
-.larger-button {
```
Has this removal affected other buttons on the site?

EChesters replied:
I checked where this was being used and I didn't find any occurrences.

tanyapowell commented on src/sass/partials/_colours.scss:
```diff
@@ -1,37 +1,40 @@
-$brand-color:      #ff6525;
-$link-hover:	   darken($brand-color, 15%);
\ No newline at end of file
+$white: 						#fff;
```
🙈 did you reformat this file? It's a bit hard to read now

EChesters replied:
I'll clean this up. For some reason it's all over the place on here but not on my Sublime.

tanyapowell commented:
This is just my opinion, but it looks a little weird __only__ have the name field with the little icon on the right hand side of the `input` box. 

![image](https://cloud.githubusercontent.com/assets/11095605/16311523/de9d8cfe-3968-11e6-990c-d2a42305bee6.png)

Could you add one to email too?

lili2311 commented on src/sass/partials/_form.scss:
```diff
@@ -1,104 +1,7 @@
-  box-shadow: 1px 1px 1px 1px rgba(0,0,0,.1) inset
-
-}
+span.help-text {
```
This is too specific, please avoid selectors like these just .help-text is sufficient to select the correct element.

tanyapowell replied:
Think bootstrap has a `.help-text`. So this is how your override it, by explicitly declaring it this way

lili2311 replied:
Since we import bootstrap first thing any styles you write afterwards naturally override it.

tanyapowell replied:
From my experience this is not always the case. Bootstrap has an annoying habit of overriding your styling if it's not explicitly instructed not to. If someone knows a way around this I'd love to know or maybe this is fixed in V4

lili2311 replied:
We imported and included bootsrap sass manually though we are not using an external stylesheet. So we have a lot more flexibility and control over what we use from bootsrap.

tanyapowell commented:
@Nandhini31 were you giving the 👍 on the whole pull request or just answering the question?

EChesters commented:
@tanyapowell The icon in the field next to name is my LastPass auto fill plugin. That isn't something I've done and it'll go away on other people's screens. 

@lili2311 The selector wasn't that specific since help text when used on a site is used everywhere. Bootstrap also has this selector and I was overriding that. 

I'm just about to change the name fields to combine them and use @tanyapowell's way of wrapping the input inside label. Will the names being combined into one cause any issues when submitting and the way we expect the data?

lili2311 commented:
`.larger-button` is being used in the projects page, please check it wont affect it

EChesters committed:
Made amendments from PR Added width for larger buttons back in for project page
Tried to wrap the label around inputs but Bootstrap styling broke making fields smaller

EChesters committed:
Fix merge conflicts in volunteer contact form

EChesters commented:
I tried @tanyapowell's suggestion but it broke the styling. This has been sidelined for now by Tanya. 

Top of volunteers form now just says name. **Will this break submitting the form?**
<img width="537" alt="screen shot 2016-06-23 at 23 20 39" src="https://cloud.githubusercontent.com/assets/2605703/16322245/ac461a42-3999-11e6-9106-7f216e5a66a3.png">

lili2311 commented on src/contact.html:
```diff
@@ -7,20 +7,32 @@
-    <label for="name">Name:</label>
-		<input type="text" name="name" placeholder="E.g. Jon Snow" required>
-    <label for="email">Email Address:</label>
-		<input type="email" name="_replyto" placeholder="E.g. jon.snow@example.com" required>
```
`name="_replyto"` is required to make the form submission work, same for other fields that have `name="_smth"` please test the form locally perhaps?  just to make sure it still works, we will need to approve the email again so ti submits locally.

EChesters replied:
The `_replyto` hasn't been touched, and is still part of the email line.

lili2311 commented on src/non-profits-contact.html:
```diff
@@ -2,31 +2,45 @@
   </div>
 </section>
-<section>
+<section class="col-lg-3 col-md-7 col-sm-9 col-xs-12 position central">
```
is there a screenshot perhaps of what it looks like now under different break points?

EChesters commented:
`.col-sm-9`
<img width="1680" alt="screen shot 2016-06-24 at 19 53 05" src="https://cloud.githubusercontent.com/assets/2605703/16347831/b52b340a-3a46-11e6-862d-7536635a8775.png">

`.col-md-7`
<img width="1680" alt="screen shot 2016-06-24 at 19 53 15" src="https://cloud.githubusercontent.com/assets/2605703/16347686/08d5dfb6-3a46-11e6-9408-b3b391190567.png">

`.col-xs-12`
<img width="1680" alt="screen shot 2016-06-24 at 19 53 19" src="https://cloud.githubusercontent.com/assets/2605703/16347710/2c790434-3a46-11e6-95ec-1fc13181f211.png">

_Please see the above for `.col-lg-3`!!_

lili2311 commented:
.col-sm-9 and .col-sm-7 both look a little too wide in my opinion, perhaps smaller width?

EChesters committed:
Issue 113 - Decreased width of forms

tanyapowell commented:
@EChesters can you send over your wireframes for any forms? If the width in your screengrab matches (or is near enough to the wireframe) then it'll be fine 😄

EChesters commented:
The prototype is here: http://echesters.co.uk/WHFNP-Website/ but I did give up on them half way through and didn't get round to doing the forms. On the wireframes these also weren't done because I hadn't done the contact pages.

tanyapowell commented:
Just been through the code and it's all good with me 👍🏿

On Wed, 6 Jul 2016 at 14:40 Elizabeth Chesters notifications@github.com
wrote:

> The prototype is here: http://echesters.co.uk/WHFNP-Website/ but I did
> give up on them half way through and didn't get round to doing the forms.
> On the wireframes these also weren't done because I hadn't done the contact
> pages.
> 
> —
> You are receiving this because you were mentioned.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/womenhackfornonprofits/whfnp/pull/113#issuecomment-230773973,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKlONUycthCrhgHU5IB3JhztbyG3RrdVks5qS7A3gaJpZM4I5On6
> .
> 
> ## 

Tanya Powell
about.me/tanyapowell
https://about.me/tanyapowell?promo=email_sig
